### PR TITLE
Destination bigquery: handle case where stream name and namespace are identical

### PR DESCRIPTION
**Pull request by edgao**
closes https://github.com/airbytehq/oncall/issues/3002.

This query:
```sql
DELETE FROM `edgao2`.`edgao2`
WHERE
  `arst` IN (
    SELECT `arst` FROM (
      SELECT `arst`, row_number() OVER (
        PARTITION BY `arst` ORDER BY `arst` DESC NULLS LAST
      ) as row_number FROM `edgao2`.`edgao2`
    )
    WHERE row_number != 1
  );
```
throws an error `Name edgao2 not found inside edgao2 at [7:37]`. The INSERT query has similar behavior. I think bigquery gets confused about whether `edgao2` refers to a dataset or a table.

Prefix the table reference with the project id (so we end up with `your_project.edgao2.edgao2`) to disambiguate.

### CODE REVIEW

File: airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/typing_deduping/BigQuerySqlGenerator.java [hunk 1]
```diff
@@ -52,14 +52,18 @@ public class BigQuerySqlGenerator implements SqlGenerator<TableDefinition> {
...
   /**
+   * @param projectId
    * @param datasetLocation This is technically redundant with {@link BigQueryDestinationHandler}
-   *        setting the query execution location, but let's be explicit since this is typically a
-   *        compliance requirement.
+   *                        setting the query execution location, but let's be explicit since this is typically a
+   *                        compliance requirement.
    */
-  public BigQuerySqlGenerator(final String datasetLocation) {
+  public BigQuerySqlGenerator(String projectId, final String datasetLocation) {
```
Comment (evantahler) [inline airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/typing_deduping/BigQuerySqlGenerator.java:65]: 👍 method signature is changed, so folks have to do it this way

Comment (evantahler) [reply airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/typing_deduping/BigQuerySqlGenerator.java:65]: There are other uses of `raw_table_id` or `final_table_id` in this file... should they all get the porject id first?

**Review by evantahler — APPROVED** [review]

**Comment by airbyte-oss-build-runner** [comment]
##  destination-bigquery test report (commit [`037f3a4898`](https://github.com/airbytehq/airbyte/pull/30640/commits/037f3a489826d78dc789d9ed8d9655ef8fbc43d1)) - ❌

⏲️  Total pipeline duration: 03mn42s 

| Step                                                                        | Result   |
|:----------------------------------------------------------------------------|:---------|
| Build connector tar                                                         | ❌        |
| Validate airbyte-integrations/connectors/destination-bigquery/metadata.yaml | ✅        |
| Connector version semver check                                              | ✅        |
| Connector version increment check                                           | ✅        |
| QA checks                                                                   | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_bigquery_streamnamenamespace/1695317749/037f3a489826d78dc789d9ed8d9655ef8fbc43d1/destination-bigquery/2.0.15/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:037f3a489826d78dc789d9ed8d9655ef8fbc43d1)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery test
```

**Comment by airbyte-oss-build-runner** [comment]
##  destination-bigquery test report (commit [`f7868591cd`](https://github.com/airbytehq/airbyte/pull/30640/commits/f7868591cdade72c9f6165a4d779479a1f6d95c3)) - ✅

⏲️  Total pipeline duration: 09mn35s 

| Step                                                                        | Result   |
|:----------------------------------------------------------------------------|:---------|
| Build connector tar                                                         | ✅        |
| Build destination-bigquery docker image for platform linux/x86_64           | ✅        |
| Java Connector Unit Tests                                                   | ✅        |
| Java Connector Integration Tests                                            | ✅        |
| Validate airbyte-integrations/connectors/destination-bigquery/metadata.yaml | ✅        |
| Connector version semver check                                              | ✅        |
| Connector version increment check                                           | ✅        |
| QA checks                                                                   | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_bigquery_streamnamenamespace/1695318585/f7868591cdade72c9f6165a4d779479a1f6d95c3/destination-bigquery/2.0.15/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:f7868591cdade72c9f6165a4d779479a1f6d95c3)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery test
```

**Comment by airbyte-oss-build-runner** [comment]
##  destination-bigquery test report (commit [`b5c85e9f65`](https://github.com/airbytehq/airbyte/pull/30640/commits/b5c85e9f657a26a5dfa84b70bbeaa344f4b862b0)) - ✅

⏲️  Total pipeline duration: 06mn23s 

| Step                                                                        | Result   |
|:----------------------------------------------------------------------------|:---------|
| Build connector tar                                                         | ✅        |
| Build destination-bigquery docker image for platform linux/x86_64           | ✅        |
| Java Connector Unit Tests                                                   | ✅        |
| Java Connector Integration Tests                                            | ✅        |
| Validate airbyte-integrations/connectors/destination-bigquery/metadata.yaml | ✅        |
| Connector version semver check                                              | ✅        |
| Connector version increment check                                           | ✅        |
| QA checks                                                                   | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_bigquery_streamnamenamespace/1695319303/b5c85e9f657a26a5dfa84b70bbeaa344f4b862b0/destination-bigquery/2.0.15/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:b5c85e9f657a26a5dfa84b70bbeaa344f4b862b0)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery test
```

**Comment by airbyte-oss-build-runner** [comment]
##  destination-duckdb test report (commit [`a4b6a63148`](https://github.com/airbytehq/airbyte/pull/30640/commits/a4b6a631481f4fd6f67ea85fe4c2be52f797531b)) - ❌

⏲️  Total pipeline duration: 01mn42s 

| Step                                                                      | Result   |
|:--------------------------------------------------------------------------|:---------|
| Connector package install                                                 | ✅        |
| Build destination-duckdb docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                                | ✅        |
| Integration tests                                                         | ✅        |
| Acceptance tests                                                          | ✅        |
| Code format checks                                                        | ✅        |
| Validate airbyte-integrations/connectors/destination-duckdb/metadata.yaml | ✅        |
| Connector version semver check                                            | ✅        |
| Connector version increment check                                         | ❌        |
| QA checks                                                                 | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_bigquery_streamnamenamespace/1695320624/a4b6a631481f4fd6f67ea85fe4c2be52f797531b/destination-duckdb/0.2.1/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:a4b6a631481f4fd6f67ea85fe4c2be52f797531b)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-duckdb test
```

**Comment by airbyte-oss-build-runner** [comment]
##  destination-bigquery test report (commit [`a4b6a63148`](https://github.com/airbytehq/airbyte/pull/30640/commits/a4b6a631481f4fd6f67ea85fe4c2be52f797531b)) - ✅

⏲️  Total pipeline duration: 09mn52s 

| Step                                                                        | Result   |
|:----------------------------------------------------------------------------|:---------|
| Build connector tar                                                         | ✅        |
| Build destination-bigquery docker image for platform linux/x86_64           | ✅        |
| Java Connector Unit Tests                                                   | ✅        |
| Java Connector Integration Tests                                            | ✅        |
| Validate airbyte-integrations/connectors/destination-bigquery/metadata.yaml | ✅        |
| Connector version semver check                                              | ✅        |
| Connector version increment check                                           | ✅        |
| QA checks                                                                   | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_bigquery_streamnamenamespace/1695320624/a4b6a631481f4fd6f67ea85fe4c2be52f797531b/destination-bigquery/2.0.15/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:a4b6a631481f4fd6f67ea85fe4c2be52f797531b)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery test
```

**Comment by octavia-approvington** [comment]
Enough! Let's do this <br/> goooooo!

**Review by octavia-approvington — APPROVED** [review]

**Pull request merged**